### PR TITLE
Added File Path and URL Support to the Vertex AI Integration

### PR DESCRIPTION
**Pull request by MinuraPunchihewa**
## Description

This PR adds the ability to pass service account keys as either a file path or URL when instantiating a Vertex AI ML engine. This is achieved using the `service_account_key_file` and `service_account_key_url` parameters respectively. The existing `service_account` parameter has also been renamed to `service_account_key_json`.

To implement the above functionality a new utility class called `GoogleServiceAccountOAuth2Manager` has been added. The existing `GoogleOAuth2Manager` has been renamed to `GoogleUserOAuth2Manager` for clarity.

Fixes https://github.com/mindsdb/mindsdb/issues/8961

## Type of change

- [X] ⚡ New feature (non-breaking change which adds functionality)

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Checklist:

- [X] My code follows the style guidelines(PEP 8) of MindsDB.
- [X] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Awesome, please check the few comments

**Comment by MinuraPunchihewa** [comment]
Hey @ZoranPandovski,
I have made the requested changes.

**Review by ZoranPandovski — APPROVED** [review]

File: mindsdb/integrations/handlers/vertex_handler/vertex_client.py [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+
 logger = log.getLogger(__name__)
 
 
 class VertexClient:
     """A class to interact with Vertex AI"""
 
-    def __init__(self, credentials_info, args_json):
-        credentials = service_account.Credentials.from_service_account_info(credentials_info)
+    def __init__(self, credentials_url, credentials_file, credentials_json, args_json):
```
Comment (ea-rus) [inline mindsdb/integrations/handlers/vertex_handler/vertex_client.py:13]: looks like all 3 methods to pass credentials required (should be passed all of them)
but in reality only one
maybe to add default values them?

**Review by ea-rus — APPROVED** [review]
The rest looks good

**Pull request merged**